src/spatialdata/_core/query/relational_query.py

```diff
@@ -462,14 +464,78 @@ class MatchTypes(Enum):
     no = "no"
 
 
-def join_sdata_spatialelement_table(
-    sdata: SpatialData,
-    spatial_element_name: str | list[str],
-    table_name: str,
-    how: str = "left",
+def _create_sdata_elements_dict_for_join(
+    sdata: SpatialData, spatial_element_name: str | list[str], table_name: str
+) -> tuple[dict[str, dict[str, Any]], AnnData]:
+    assert sdata.tables.get(table_name), f"No table with `{table_name}` exists in the SpatialData object."
+    table = sdata.tables[table_name]
+
+    elements_dict: dict[str, dict[str, Any]] = defaultdict(lambda: defaultdict(dict))
+    for name in spatial_element_name:
+        if name in sdata.tables:
+            warnings.warn(
+                f"Table: `{name}` given in spatial_element_names cannot be "
+                f"joined with a table using this function.",
+                UserWarning,
+                stacklevel=2,
+            )
+        elif name in sdata.images:
+            warnings.warn(
+                f"Image: `{name}` cannot be joined with a table",
+                UserWarning,
+                stacklevel=2,
+            )
+        else:
+            element_type, _, element = sdata._find_element(name)
+            elements_dict[element_type][name] = element
+    return elements_dict, table
+
+
+def _create_elements_dict_for_join(
```

Comment:
This function, except for the warnings part, can be replaced by `SpatialData.init_from_elements()`.

Reply:
I did this.